src/core/net/nat64_translator.cpp

```diff
@@ -510,6 +510,17 @@ Error Translator::SetIp4Cidr(const Ip4::Cidr &aCidr)
     return err;
 }
 
+void Translator::ClearIp4Cidr(void)
+{
+    mIp4Cidr.Clear();
+    mAddressMappingPool.FreeAll();
+    mActiveAddressMappings.Clear();
+    mIp4AddressPool.Clear();
+
+    UpdateState();
+    Get<Notifier>().Signal(kEventNat64TranslatorStateChanged);
```

Comment:
Should we check if there is any actual change to state before signalling (e.g. if Cidr was already cleared, I expect calling `ClearIp4Cidr()` would not change state?

Reply:
Actually, looking at the code, I think `UpdateState()` already checks and signals properly. So I think we can remove the ` Get<Notifier>().Signal(kEvent...)` here?

Reply:
Done. Thanks for the suggestion!